File: app/src/main/java/com/past/music/activity/ToolBarActivity.java
package com.past.music.activity;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.past.music.pastmusic.R;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/3/5 下午7:05
 * 描述：带有Toolbar的Activity抽象基类
 * 备注：Copyright © 2010-2017. gaojin All rights reserved.
 * =======================================================
 */

public abstract class ToolBarActivity extends KtBaseActivity {

    @BindView(R.id.title)
    protected TextView mTitle;

    @BindView(R.id.back)
    protected ImageView mBack;

    @BindView(R.id.base_toolbar)
    RelativeLayout relativeLayout;

    @OnClick(R.id.back)
    void back() {
        finish();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
    }

    protected abstract int getLayoutId();

    @Override
    public void setTitle(CharSequence title) {
        mTitle.setText(title);
    }

    @Override
    public void setStatusBar() {
    }
}

File: app/src/main/java/com/se/music/activity/ToolBarActivity.kt
package com.se.music.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.se.music.R

/**
 * Creator：gaojin
 * date：2017/11/6 下午8:32
 */
abstract class ToolBarActivity : AppCompatActivity() {
    private var mToolBar: Toolbar? = null
    private var mTitle: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val inflater = LayoutInflater.from(this)
        val rootView = inflater.inflate(R.layout.fragment_base, null) as LinearLayout
        rootView.addView(createContentView(inflater, rootView))
        setContentView(rootView)
        mToolBar = findViewById(R.id.base_toolbar)
        mTitle = findViewById(R.id.toolbar_title)
        setSupportActionBar(mToolBar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowTitleEnabled(false)
    }

    protected abstract fun createContentView(inflater: LayoutInflater, rootView: ViewGroup): View

    override fun setTitle(title: CharSequence) {
        mTitle!!.text = title
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            android.R.id.home -> finish()
        }
        return true
    }

}